apps/codeforafrica/src/lib/api.netlify-cms/utils.js

```diff
@@ -0,0 +1,28 @@
+import fs from "fs";
+import { join } from "path";
+
+import matter from "gray-matter";
+import { marked } from "marked";
+
+export function getCollectionSlugs(collectionDir) {
+  return fs.readdirSync(collectionDir);
+}
+
+export function getCollectionBySlug(collectionDir, slug, fields) {
+  const realSlug = slug.replace(/\.md$/, "");
+  const fullPath = join(collectionDir, `${realSlug}.md`);
+  const fileContents = fs.readFileSync(fullPath, "utf8");
+  const { data, content } = matter(fileContents);
+
+  const items = fields?.reduce((acc, curr) => {
+    if (curr === "content") {
+      acc.content = marked(content);
+    } else if (curr === "slug") {
+      acc.slug = realSlug;
+    } else {
+      acc[curr] = data[curr] || null;
+    }
+    return acc;
+  }, {});
+  return { items, data }; // return data which can be used as default incase of no fields
```

Comment:
I'm still not 💯 on this: Say no fields is requested, we return `data`. Cool... the problem however is `data` doesn't include `content` or `realSlug` (could be using slug with `.md`). So, in the end, when you don't specify fields, you really still don't get everything.

Reply:
For me, I still think there should be a default behavior which as you had said earlier is returning all the data we have read in the markdown file. We then let the consumer decide on what to do with the data. So on this, I would suggest just including `content` into the returned data which is all 'data' we got from the MD.

Reply:
Cool, then lets return _all_ data, including `content` and `slug`.